tests/bdd/test_volume_create.py

```diff
@@ -0,0 +1,195 @@
+    """a control plane, Mayastor instances and a pool."""
+    docker_client = docker.from_env()
+
+    # The control plane comprises the core agents, rest server and etcd instance.
+    for component in ["core", "rest", "etcd"]:
+        try:
+            container = docker_client.containers.get(component)
+        except docker.errors.NotFound as exc:
+            raise Exception("{} container not found", component)
+        else:
+            container_state = container.attrs["State"]
+            if container_state["Status"] != "running":
+                raise Exception("{} container not running", component)
+
+    # Check all Mayastor instances are running
+    try:
+        mayastors = docker_client.containers.list(
+            all=True, filters={"name": "mayastor"}
+        )
+    except docker.errors.NotFound:
+        raise Exception("No Mayastor instances")
+
+    for mayastor in mayastors:
+        container_state = mayastor.attrs["State"]
+        if container_state["Status"] != "running":
+            raise Exception("{} container not running", mayastor.attrs["Name"])
+
+    # Check for a pool
+    pool = common.get_pools_api().get_pool(POOL_UUID)
+    assert pool.id == POOL_UUID
+
+
+@given("a request for a volume")
+def a_request_for_a_volume(create_request):
+    """a request for a volume."""
+    policy = {"self_heal": False, "topology": None}
+    topology = {"explicit": None, "labelled": None}
+    request = CreateVolumeBody(policy, NUM_VOLUME_REPLICAS, VOLUME_SIZE, topology)
+    create_request[CREATE_REQUEST_KEY] = request
+
+
+@when("the number of suitable pools is less than the number of desired volume replicas")
+def the_number_of_suitable_pools_is_less_than_the_number_of_desired_volume_replicas(
+    create_request,
+):
+    """the number of suitable pools is less than the number of desired volume replicas."""
+    # Delete the pool so that there aren't enough
+    pools_api = common.get_pools_api()
+    pools_api.del_pool(POOL_UUID)
+    num_pools = len(pools_api.get_pools())
+    num_volume_replicas = create_request[CREATE_REQUEST_KEY]["replicas"]
+    assert num_pools < num_volume_replicas
+
+
+@when(
+    "the number of volume replicas is less than or equal to the number of suitable pools"
+)
+def the_number_of_volume_replicas_is_less_than_or_equal_to_the_number_of_suitable_pools(
+    create_request,
+):
+    """the number of volume replicas is less than or equal to the number of suitable pools."""
+    num_pools = len(common.get_pools_api().get_pools())
+    num_volume_replicas = create_request[CREATE_REQUEST_KEY]["replicas"]
+    assert num_volume_replicas <= num_pools
+
+
+@when("there are no available Mayastor instances")
+def there_are_no_available_mayastor_instances():
+    """there are no available Mayastor instances."""
+    # Kill mayastor instance
+    docker_client = docker.from_env()
+    container = docker_client.containers.get("mayastor")
+    container.stop()
```

Comment:
rather than the sleep, can we just do container.kill() ?

Reply:
Done.